webdriver/tests/bidi/browsing_context/locate_nodes/invalid.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+import webdriver.bidi.error as error
+
+pytestmark = pytest.mark.asyncio
+
+
+@pytest.mark.parametrize("value", [False, 42, {}, []])
+async def test_params_context_invalid_type(bidi_session, inline, top_context, value):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=value, locator={ "type": "css", "value": "div" }
+        )
+
+
+async def test_params_context_invalid_value(bidi_session, inline, top_context):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context="foo", locator={ "type": "css", "value": "div" }
+        )
+
+
+@pytest.mark.parametrize("value", [False, 42, {}, []])
+async def test_params_locator_invalid_type(bidi_session, inline, top_context, value):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"], locator={ "type": value, "value": "div" }
+        )
+
+
+async def test_params_locator_invalid_value(bidi_session, inline, top_context):
+    url = inline("""<div>foo<span><strong>bar</strong></span><span>BAR</span>baz</div>""")
+    await bidi_session.browsing_context.navigate(
+        context=top_context["context"], url=url, wait="complete"
+    )
+
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.locate_nodes(
+            context=top_context["context"], locator={ "type": "invalid", "value": "div" }
```

Comment:
I would suggest to parameterize this test as well so that we can test an empty string as well as variations of `csss` and `xpaths` as well. The `innerText` locator is a bit more complex so we should test it on its own.

Reply:
I've separated this out into different (parameterized) tests. One for an invalid locator type, the other for invalid locator values, as they return different errors.